src/naive/date.rs

```diff
@@ -2673,22 +2724,22 @@ mod tests {
             assert_eq!(lhs.checked_add_days(rhs), sum);
         }
 
-        check((2014, 1, 1), Days::new(0), Some((2014, 1, 1)));
+        check((2014, 1, 1), Days::from_u32(0), Some((2014, 1, 1)));
```

Comment:
For backwards compatibility checks, could you add some tests cases (or statements within a test case) to compare `Days::new` and `Days::from_u32` result in the same values?

I know `new` is the "old way", but wouldn't hurt to make sure it's still working.